liquibase-standard/src/main/java/liquibase/snapshot/jvm/SequenceSnapshotGenerator.java

```diff
@@ -266,4 +263,16 @@ protected SqlStatement getSelectSequenceStatement (Schema schema, Database datab
             throw new UnexpectedLiquibaseException("Don't know how to query for sequences on " + database);
         }
     }
+
+    private static RawParameterizedSqlStatement getMSSQLQuery(Schema schema) {
+        String sql = "SELECT SEQUENCE_NAME, START_VALUE, MINIMUM_VALUE AS MIN_VALUE, " +
+                "MAXIMUM_VALUE AS MAX_VALUE, INCREMENT AS INCREMENT_BY, CYCLE_OPTION AS WILL_CYCLE, " +
+                // If we have a decimal sequence we want to include the precision with the data type.
+                // This is not necessary with types like bigint. Previously we did not include the data type
+                // in our query which would create the sequence with data type bigint. So we normalize all other
+                // non-decimal seq_type values to null.
+                "IIF(DATA_TYPE = 'decimal', DATA_TYPE + '(' + CAST(NUMERIC_PRECISION AS VARCHAR) + ')', NULL) AS SEQ_TYPE " +
```

Comment:
I can also capture this `NUMERIC_PRECISION` column in the sequence class and do the concat in one of our generators, but this felt less intrusive.